controller/src/controller/config.py

```diff
@@ -86,6 +93,54 @@ def _reject_reserved_paths(v: str) -> str:
     return v
 
 
+class _VersionPydanticAnnotation:
```

Comment:
This seems excessive just to be able to store the parsed version number, which I don't think has a lot of benefit. If you keep the type as `str` you can drop all of this. If you want to validate the version, just add a functional validator that checks that you can parse the string with semver.

Reply:
Yeah, while I was working on this over the weekend (not having seen the review yet) I came to basically that conclusion and just added a string-to-Version validator.